app/Services/CSVImportReader.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Services;
+
+use Illuminate\Support\LazyCollection;
+use App\Exceptions\ImportParserException;
+
+class CSVImportReader
+{
+    public function lines(string $path): LazyCollection
+    {
+        $lines = LazyCollection::make(function () use ($path) {
+            $file = fopen($path, 'r');
+
+            while ($data = fgetcsv($file)) {
+                yield $data;
+            }
+        });
+
+        $allowedKeys = config('imports.upload.column_keys');
+        $columnKeys = $lines->get(0);
+
+        if (count(array_diff($allowedKeys, $columnKeys)) !== 0) {
```

Comment:
this check will not fail if all required header fields are there, but columns are in the wrong order, right? Is that a case we would want to reject?

Reply:
I don't think so, because if they are in the wrong order then the file would fail validation, so people will have to make sure that the order of each row matches the order of the columns.